api/src/main/java/org/openmrs/module/reporting/cohort/definition/evaluator/DrugOrderCohortDefinitionEvaluator.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.reporting.cohort.definition.evaluator;
+
+import org.openmrs.annotation.Handler;
+
+import java.util.List;
+import java.util.Date;
+
+import org.openmrs.Cohort;
+import org.openmrs.DrugOrder;
+import org.openmrs.Order.Action;
+import org.openmrs.module.reporting.cohort.definition.CohortDefinition;
+import org.openmrs.module.reporting.evaluation.EvaluationContext;
+import org.openmrs.module.reporting.evaluation.querybuilder.HqlQueryBuilder;
+import org.openmrs.module.reporting.evaluation.service.EvaluationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.openmrs.module.reporting.cohort.EvaluatedCohort;
+import org.openmrs.module.reporting.cohort.definition.DrugOrderCohortDefinition;
+import org.openmrs.module.reporting.common.ObjectUtil;
+
+/**
+ * Evaluates a DrugOrderCohortDefinition and produces a cohort
+ */
+@Handler(supports={DrugOrderCohortDefinition.class})
+public class DrugOrderCohortDefinitionEvaluator implements CohortDefinitionEvaluator {
+
+  @Autowired
+  EvaluationService evaluationService;
+
+  /**
+   * Default constructor
+   */
+  public DrugOrderCohortDefinitionEvaluator(){}
+
+  public EvaluatedCohort evaluate(CohortDefinition cohortDefinition, EvaluationContext context){
+    DrugOrderCohortDefinition cd = (DrugOrderCohortDefinition) cohortDefinition;
+		context = ObjectUtil.nvl(context, new EvaluationContext());
+
+		HqlQueryBuilder q = new HqlQueryBuilder();
+		q.select("do.patient.patientId");
+		q.from(DrugOrder.class, "do");
+
+		q.wherePatientIn("do.patient.patientId", context);
+		
+		// For a drug to be active, it must have a valid activated date and a stoppedDate of null value or greater than now?
+		if (cd.getOnlyCurrentlyActive()) {
+			q.whereNotNull("do.dateActivated");
```

Comment:
Active as of when - now?  We like want this to be "activeOnOrBefore" and/or "activeOnOrAfter" dates.  We also might want to consider following the model we took with the Program Enrollment definitions, where we have separate definitions for "InProgramCohortDefinition" vs. "ProgramEnrollmentCohortDefinition".